multisrc/overrides/animestream/mykdrama/src/MyKdrama.kt

```diff
@@ -0,0 +1,121 @@
+package eu.kanade.tachiyomi.animeextension.fr.mykdrama
+
+import eu.kanade.tachiyomi.animeextension.fr.mykdrama.MyKdramaFilters.CountryFilter
+import eu.kanade.tachiyomi.animeextension.fr.mykdrama.MyKdramaFilters.GenresFilter
+import eu.kanade.tachiyomi.animeextension.fr.mykdrama.MyKdramaFilters.OrderFilter
+import eu.kanade.tachiyomi.animeextension.fr.mykdrama.MyKdramaFilters.StatusFilter
+import eu.kanade.tachiyomi.animeextension.fr.mykdrama.MyKdramaFilters.TypeFilter
+import eu.kanade.tachiyomi.animeextension.fr.mykdrama.MyKdramaFilters.getSearchParameters
+import eu.kanade.tachiyomi.animesource.model.AnimeFilter
+import eu.kanade.tachiyomi.animesource.model.AnimeFilterList
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.lib.doodextractor.DoodExtractor
+import eu.kanade.tachiyomi.lib.okruextractor.OkruExtractor
+import eu.kanade.tachiyomi.lib.uqloadextractor.UqloadExtractor
+import eu.kanade.tachiyomi.lib.vudeoextractor.VudeoExtractor
+import eu.kanade.tachiyomi.multisrc.animestream.AnimeStream
+import eu.kanade.tachiyomi.multisrc.animestream.AnimeStreamFilters
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.util.asJsoup
+import eu.kanade.tachiyomi.util.parallelCatchingFlatMapBlocking
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class MyKdrama : AnimeStream(
+    "fr",
+    "MyKdrama",
+    "https://mykdrama.co",
+) {
+    override val animeListUrl = "$baseUrl/drama"
+
+    override val dateFormatter by lazy {
+        SimpleDateFormat("MMMM dd, yyyy", Locale("fr"))
+    }
+
+    // =============================== Search ===============================
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val params = getSearchParameters(filters)
+        return if (query.isNotEmpty()) {
+            GET("$baseUrl/page/$page/?s=$query")
+        } else {
+            val queryParams = params.run { listOf(genres) }
+                .filter(String::isNotBlank)
+                .joinToString("&")
+            val url = "$animeListUrl/?$queryParams".toHttpUrl().newBuilder()
```

Comment:
Wouldn't it be better to add the query parameters in the builder, so that special characters are properly handled?

Reply:
tbh i merely reproduced what AsyaAnimeleri is doing

Reply:
oh ye, it's not doable here because genres (and country which I fix in the commit) is processed into a query array like `country[]=korea&country[]=china` (that syntax is so ugly tf??)
https://github.com/aniyomiorg/aniyomi-extensions/blob/2db273348929ce9369226f20a1bc4f40f7c71650/multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/animestream/AnimeStreamFilters.kt#L31-L40